### PR TITLE
clang-tidy: Assorted source fixes

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

**Comment by phlax** [comment]
cc @mathetake

File: source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc [hunk 1]
```diff
@@ -129,7 +129,7 @@ void TimerTokenBucket::onFillTimer(uint64_t refill_counter, double factor) {
   // descriptor refill interval over the global refill interval. For example,
   // if the descriptor refill interval is 150ms and the global refill
   // interval is 50ms, this descriptor is refilled every 3rd call.
-  if (refill_counter % multiplier_ != 0) {
+  if (multiplier_ != 0 && refill_counter % multiplier_ != 0) {
```
Comment (ggreenway) [inline source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc:132]: I don't think this case can happen. If you add `ASSERT(multiplier_ != 0);` to the previous line does it fix the clang-tidy error? And maybe also add that assert in the constructor `TimerTokenBucket::TimerTokenBucket`.

**Review by ggreenway — CHANGES_REQUESTED** [review]
/wait

File: source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc (hunk 1 above)
Comment (phlax) [reply source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc:132]: will check ...

Comment (phlax) [reply source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc]: seems to work with the assert

**Review by ggreenway — APPROVED** [review]

**Pull request merged**